### PR TITLE
Docs: Add LSP-clangd and LSP-terraform to Language Servers

### PR DESCRIPTION
**Pull request by rchl**
Docs changes
- Add LSP-clangd and LSP-terraform to Language Servers
- Remove C++ guide as it's covered in more concise form by [LSP-clangd's readme](https://github.com/sublimelsp/LSP-clangd/pull/1)
- Move /guides/client_configuration to /client_configuration as it's the only "guide" at this point
- Set up redirects so that old URLs still work/redirect

Related to https://github.com/sublimelsp/LSP-clangd/pull/1

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
       - uses: actions/setup-python@v2
         with:
           python-version: 3.x
-      - run: pip install mkdocs-material
+      - run: pip install mkdocs-material mkdocs-redirects
```
Comment (predragnikolic) [inline .github/workflows/docs.yml:17]: the `LSP/docs/README.md` also needs to be updated to include `mkdocs-redirects`:
```
Install Material Theme - https://squidfunk.github.io/mkdocs-material/getting-started/#installation

pip install mkdocs mkdocs-material mkdocs-redirects

```

File: docs/mkdocs.yml [hunk 2]
```diff
@@ -37,14 +44,12 @@ markdown_extensions:
 
 nav:
     - Getting Started: index.md
-    - Features: features.md
     - Language Servers: language_servers.md
+    - Features: features.md
```
Comment (predragnikolic) [inline docs/mkdocs.yml:48]: nice change!

**Review by predragnikolic — APPROVED** [review]
Tested Redirects :heavy_check_mark: 

Just notices this small thing
https://github.com/sublimelsp/LSP/pull/2096/files#r1002673654

**Comment by predragnikolic** [comment]
For some reason the github action to build the docs wasn't run.

**Comment by rchl** [comment]
Don't think those deprecation warnings are relevant to the issue.
Deployment is happening from the `gh-pages` branch: https://github.com/sublimelsp/LSP/actions/runs/3354949410/jobs/5558959354
I'll try to restart the failed job there.

**Comment by rchl** [comment]
https://github.com/sublimelsp/LSP/settings/environments/51262986/edit didn't white list gh-pages branch. Should be good now.

**Comment by predragnikolic** [comment]
You fixed it :D how?

**Pull request merged**